trunk/src/meshmod/close_mesh.F90

```diff
@@ -147,24 +156,27 @@ subroutine close_mesh()
             else
 !           ...-------------------------------------------------------------------
 !           ...Option 1: do minimum refinement that is necessary
-               !call find_element_closing_ref(ntype,kreff,krefe, kref)
+!               call find_element_closing_ref(ntype,kreff,krefe, kref)
 !           ...-------------------------------------------------------------------
 !           ...Option 2: always ask for isotropic refinement
                call get_isoref(mdle, kref)
 !           ...-------------------------------------------------------------------
 !           ...Option 3: always ask for radial (xy) refinement (FIBER LASER)
-               !select case (NODES(mdle)%ntype)
-               !   case(MDLB); kref = 110
-               !   case(MDLP); kref = 10
-               !end select
+!               select case (NODES(mdle)%ntype)
+!                  case(MDLB); kref = 110
+!                  case(MDLP); kref = 10
+!               end select
 !           ...-------------------------------------------------------------------
             endif
             list(1,i) = mdle
             list(2,i) = kref
+!$OMP END CRITICAL
          endif
       enddo
 !$OMP END DO
 !$OMP END PARALLEL
+!
+      call bitvisit_finalize
```

Comment:
Instead of doing `bitvisit_init` and `bitvisit_finalize` inside the outer do-loop, would it make more sense to init and finalize outside the loop, and do `bivisit_reset` here?

Reply:
Refresh needs it too, that's why it's done this way here. It's a bit sloppy, I'd like to implement this class more like `index_map` (where you can have multiple instances)

Reply:
To speed this up here, because it is allocated and deallocated twice per iteration, you could make a minor change to refresh that check if `bitvisit` had previously been initialized (with size NRNODS) and then only `bitvisit_reset` and skip `bitvisit_finalize` in refresh.